nupic/research/frameworks/pytorch/cifar_experiment.py

```diff
@@ -195,81 +233,85 @@ def model_save(self, checkpoint_dir):
-    # torch.save(self.model, checkpoint_path)
+    # torch.save(self.model.state_dict(), checkpoint_path)
     checkpoint_path = os.path.join(checkpoint_dir, self.model_filename)
-    torch.save(self.model.state_dict(), checkpoint_path)
+
+    # Use the slow method if filename ends with .pt
+    if checkpoint_path.endswith(".pt"):
+      torch.save(self.model, checkpoint_path)
+    else:
+      torch.save(self.model.state_dict(), checkpoint_path)
+
     return checkpoint_path
 
 
   def model_restore(self, checkpoint_path):
     """
-    :param checkpoint_path: Loads model from this checkpoint path
-    :return:
-    """
-    # self.model = torch.load(checkpoint_path, map_location=self.device)
-    checkpoint_file = os.path.join(checkpoint_path, self.model_filename)
-    self.model.load_state_dict(torch.load(checkpoint_file, map_location=self.device))
+    :param checkpoint_path: Loads model from this checkpoint path.
+    If path is a directory, will append the parameter model_filename
 
-
-  def _createTestLoaders(self, noise_values):
-    """
-    Create a list of data loaders, one for each noise value
     """
-    print("Creating test loaders for noise values:", noise_values)
-    loaders = []
-    for noise in noise_values:
-
-      transform_noise_test = transforms.Compose([
-        transforms.ToTensor(),
-        transforms.Normalize((0.4914, 0.4822, 0.4465), (0.2023, 0.1994, 0.2010)),
-        RandomNoise(noise,
-                    highValue=0.5 + 2 * 0.20,
-                    lowValue=0.5 - 2 * 0.2,
-                    ),
-      ])
-
-      testset = datasets.CIFAR10(root=self.data_dir,
-                                 train=False,
-                                 transform=transform_noise_test)
-      loaders.append(
-        DataLoader(testset, batch_size=self.test_batch_size, shuffle=False)
-      )
+    print("loading from", checkpoint_path)
+    if os.path.isdir(checkpoint_path):
+      checkpoint_file = os.path.join(checkpoint_path, self.model_filename)
+    else:
+      checkpoint_file = checkpoint_path
 
-    return loaders
+    # Use the slow method if filename ends with .pt
+    if checkpoint_file.endswith(".pt"):
+      self.model = torch.load(checkpoint_file, map_location=self.device)
+    else:
+      self.model.load_state_dict(torch.load(checkpoint_file, map_location=self.device))
 
 
   def _createTinySparseModel(self):
+
+    # Figure out some numbers we'll need later on.
     prev_w = self.w
     cnn_output_len = []
-    for i, (ks,ch) in enumerate(zip(self.kernel_size, self.out_channels)):
-      cnn_w = CNNSize(prev_w, ks) // 2
+    padding = []
+    for i, (kernel_size, ch) in enumerate(zip(self.cnn_kernel_sizes,
+                                              self.cnn_out_channels)):
+      if kernel_size == 3:
+        padding.append(1)
+      else:
+        padding.append(2)
+      cnn_w = CNNSize(prev_w, kernel_size, padding=padding[-1]) // 2
```

Comment:
Do you still need to compute the `cnn_output_len` for every layer?

Reply:
No I don’t. I’ll change it another PR as I want to change this code quite a bit.